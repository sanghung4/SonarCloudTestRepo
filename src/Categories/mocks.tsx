import { DocumentNode } from 'graphql';
import {
  ProductCategoriesDocument,
  ProductCategoriesQuery
} from 'generated/graphql';

export const success: MockCategory = {
  request: {
    query: ProductCategoriesDocument,
    variables: {
      engines: 'plumbing_hvac'
    }
  },
  result: {
    data: {
      productCategories: {
        categories: [
          {
            name: 'Faucets, Fixtures & Appliances',
            children: [
              {
                name: 'TBC',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Faucets, Showers & Bathroom Accessories',
                children: [
                  {
                    name: 'Showers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Bathroom Accessories',
                    __typename: 'Category'
                  },
                  {
                    name: 'Tub & Shower',
                    __typename: 'Category'
                  },
                  {
                    name: 'Bathroom Faucets',
                    __typename: 'Category'
                  },
                  {
                    name: 'Kitchen Faucets',
                    __typename: 'Category'
                  },
                  {
                    name: 'Tub Filler',
                    __typename: 'Category'
                  },
                  {
                    name: 'Spare Parts',
                    __typename: 'Category'
                  },
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Kitchen Accessories',
                    __typename: 'Category'
                  },
                  {
                    name: 'CATCLEAN',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Appliances',
                children: [
                  {
                    name: 'Refrigerators And Freezers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Hoods and Vents',
                    __typename: 'Category'
                  },
                  {
                    name: 'Outdoor Grills and BBQ',
                    __typename: 'Category'
                  },
                  {
                    name: 'Ranges and Stoves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Microwaves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Garbage Disposals',
                    __typename: 'Category'
                  },
                  {
                    name: 'Appliance Accessories',
                    __typename: 'Category'
                  },
                  {
                    name: 'Dishwashers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Dryers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Ovens',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Toilets, Urinals, Sinks & Flush Valves',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Toilet & Flush Valve Spare Parts',
                    __typename: 'Category'
                  },
                  {
                    name: 'CATCLEAN',
                    __typename: 'Category'
                  },
                  {
                    name: 'Manual Flush Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Spare Parts',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Fans & Ventilation',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Tools',
                children: [],
                __typename: 'Category'
              }
            ],
            __typename: 'Category'
          },
          {
            name: 'Plumbing Installation, Tools, Hardware & Safety',
            children: [
              {
                name: 'Plumbing Installation',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Pipe Clamps',
                    __typename: 'Category'
                  },
                  {
                    name: 'Pipe Insulation',
                    __typename: 'Category'
                  },
                  {
                    name: 'Faucet and Toilet Parts',
                    __typename: 'Category'
                  },
                  {
                    name: 'Gaskets',
                    __typename: 'Category'
                  },
                  {
                    name: 'Pipe Hangers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Riser',
                    __typename: 'Category'
                  },
                  {
                    name: 'Insulation',
                    __typename: 'Category'
                  },
                  {
                    name: 'Anchors and Settings',
                    __typename: 'Category'
                  },
                  {
                    name: 'Compression Fittings',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Tools',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Power Tool',
                    __typename: 'Category'
                  },
                  {
                    name: 'Brushes',
                    __typename: 'Category'
                  },
                  {
                    name: 'Tool',
                    __typename: 'Category'
                  },
                  {
                    name: 'Shovels',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'TBC',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Hardware',
                children: [
                  {
                    name: 'Screws',
                    __typename: 'Category'
                  },
                  {
                    name: 'Washers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Nuts',
                    __typename: 'Category'
                  },
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Bolts',
                    __typename: 'Category'
                  },
                  {
                    name: 'Pipe Hangers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Nails',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Safety',
                children: [
                  {
                    name: 'Gloves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Glasses',
                    __typename: 'Category'
                  },
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Hard Hat',
                    __typename: 'Category'
                  },
                  {
                    name: 'Vest',
                    __typename: 'Category'
                  },
                  {
                    name: 'Shoe Covers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Beanie',
                    __typename: 'Category'
                  },
                  {
                    name: 'Ear Plugs',
                    __typename: 'Category'
                  },
                  {
                    name: 'Face Shield',
                    __typename: 'Category'
                  },
                  {
                    name: 'Knee Pads',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Tools, Hardware & Safety',
                children: [
                  {
                    name: 'CATCLEAN',
                    __typename: 'Category'
                  },
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Brushes',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Appliances',
                children: [
                  {
                    name: 'Garbage Disposals',
                    __typename: 'Category'
                  },
                  {
                    name: 'Dishwashers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Washing Machines',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Pipe Accessories',
                children: [
                  {
                    name: 'Pipe Insulation',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'HVAC',
                children: [
                  {
                    name: 'CATCLEAN',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Discontinued',
                children: [],
                __typename: 'Category'
              }
            ],
            __typename: 'Category'
          },
          {
            name: 'Pipe & Fittings',
            children: [
              {
                name: 'Metal Pipe & Fittings',
                children: [
                  {
                    name: 'Copper Tube & Fittings',
                    __typename: 'Category'
                  },
                  {
                    name: 'Steel Pipe & Fittings',
                    __typename: 'Category'
                  },
                  {
                    name: 'Cast Iron Pipe & Fittings',
                    __typename: 'Category'
                  },
                  {
                    name: 'Brass Pipe & Fittings',
                    __typename: 'Category'
                  },
                  {
                    name: 'Ductile Iron Pipe & Fittings',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Plastic Pipe & Fittings - Supply',
                children: [
                  {
                    name: 'PEX Pipe & Fittings',
                    __typename: 'Category'
                  },
                  {
                    name: 'CPVC Pipe & Fittings',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Plastic Pipe & Fittings - Drainage',
                children: [
                  {
                    name: 'PVC Pipe & Fittings',
                    __typename: 'Category'
                  },
                  {
                    name: 'ABS Pipe & Fittings',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'TBC',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Plumbing Installation',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              }
            ],
            __typename: 'Category'
          },
          {
            name: 'Heating & Cooling',
            children: [
              {
                name: 'Air Distribution',
                children: [
                  {
                    name: 'Sheet Metal Pipe & Fittings',
                    __typename: 'Category'
                  },
                  {
                    name: 'Grills, Registers & Diffusers',
                    __typename: 'Category'
                  },
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Flex Duct',
                    __typename: 'Category'
                  },
                  {
                    name: 'Sheet Metal Duct Pipe and Fittings',
                    __typename: 'Category'
                  },
                  {
                    name: 'Access & Inspection Panels',
                    __typename: 'Category'
                  },
                  {
                    name: 'CEILING RADIATION DAMPERS',
                    __typename: 'Category'
                  },
                  {
                    name: 'Grills, Registers, & Diffusers',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Repair & Replacement Parts',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Capacitors',
                    __typename: 'Category'
                  },
                  {
                    name: 'OEM Parts',
                    __typename: 'Category'
                  },
                  {
                    name: 'Contactors, Relays, Transformers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Motors',
                    __typename: 'Category'
                  },
                  {
                    name: 'Driers & Sight Glasses',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'TBC',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Residential Equipment',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'AC Condensers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Residential Accessories',
                    __typename: 'Category'
                  },
                  {
                    name: 'Air Handlers',
                    __typename: 'Category'
                  },
                  {
                    name: 'HP Condensers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Coils',
                    __typename: 'Category'
                  },
                  {
                    name: 'Furnaces',
                    __typename: 'Category'
                  },
                  {
                    name: 'Packaged Systems',
                    __typename: 'Category'
                  },
                  {
                    name: 'furnaces',
                    __typename: 'Category'
                  },
                  {
                    name: 'Condensers',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Commercial Equipment',
                children: [
                  {
                    name: 'Commercial Packaged Systems',
                    __typename: 'Category'
                  },
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Commercial Air Handlers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Geothermal',
                    __typename: 'Category'
                  },
                  {
                    name: 'Packaged Systems',
                    __typename: 'Category'
                  },
                  {
                    name: 'Residential Accessories',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Installation Supplies',
                children: [
                  {
                    name: 'Tapes, Silicon, & Adhesives',
                    __typename: 'Category'
                  },
                  {
                    name: 'Electrical',
                    __typename: 'Category'
                  },
                  {
                    name: 'Insulation',
                    __typename: 'Category'
                  },
                  {
                    name: 'CATCLEAN',
                    __typename: 'Category'
                  },
                  {
                    name: 'Condensate Management',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Fans & Ventilation',
                children: [
                  {
                    name: 'Gas Vent Pipe & Fittings',
                    __typename: 'Category'
                  },
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Ventilation Fans',
                    __typename: 'Category'
                  },
                  {
                    name: 'CATCLEAN',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'IAQ',
                children: [
                  {
                    name: 'Air Filters',
                    __typename: 'Category'
                  },
                  {
                    name: 'Air Cleaners',
                    __typename: 'Category'
                  },
                  {
                    name: 'Dehumidifiers',
                    __typename: 'Category'
                  },
                  {
                    name: 'CATCLEAN',
                    __typename: 'Category'
                  },
                  {
                    name: 'Humidifiers',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Controls, Sensors, & Zoning',
                children: [
                  {
                    name: 'Thermostats',
                    __typename: 'Category'
                  },
                  {
                    name: 'Zoning',
                    __typename: 'Category'
                  },
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Ductless',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Condensers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Air Handlers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Residential Accessories',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              }
            ],
            __typename: 'Category'
          },
          {
            name: 'Hot Water, Valves, Irrigation & Pumps',
            children: [
              {
                name: 'TBC',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Water Heater Parts & Accessories',
                children: [
                  {
                    name: 'Venting',
                    __typename: 'Category'
                  },
                  {
                    name: 'Water Heater Accessories',
                    __typename: 'Category'
                  },
                  {
                    name: 'Commercial Conversion Kits',
                    __typename: 'Category'
                  },
                  {
                    name: 'Burners',
                    __typename: 'Category'
                  },
                  {
                    name: 'Heating Element',
                    __typename: 'Category'
                  },
                  {
                    name: 'Gas Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Switches',
                    __typename: 'Category'
                  },
                  {
                    name: 'Expansion Tanks',
                    __typename: 'Category'
                  },
                  {
                    name: 'Instantaneous',
                    __typename: 'Category'
                  },
                  {
                    name: 'Electronic Boards',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Specialty Plumbing Valves',
                children: [
                  {
                    name: 'Hydrants/Flow Control Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Pressure Reducing Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Wye Strainers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Valve Kits',
                    __typename: 'Category'
                  },
                  {
                    name: 'Relief Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Thermostatic Mixing Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Hose Bibbs',
                    __typename: 'Category'
                  },
                  {
                    name: 'Dielectric Unions',
                    __typename: 'Category'
                  },
                  {
                    name: 'Sillcocks',
                    __typename: 'Category'
                  },
                  {
                    name: 'Automatic Control Valves',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Import Valves',
                children: [
                  {
                    name: 'Ball Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Check Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Gate Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Balancing Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Butterfly Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Valve Accessories',
                    __typename: 'Category'
                  },
                  {
                    name: 'Globe Valves',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Domestic Valves',
                children: [
                  {
                    name: 'Ball Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Check Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Butterfly Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Gate Valves',
                    __typename: 'Category'
                  },
                  {
                    name: 'Valve Accessories',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Pumps',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Parts and Accessories',
                    __typename: 'Category'
                  },
                  {
                    name: 'CATCLEAN',
                    __typename: 'Category'
                  },
                  {
                    name: 'Recirculating Pumps',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Backflow Preventers & Accessories',
                children: [
                  {
                    name: 'Backflow Preventers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Backflow Preventer Vacuum Breakers',
                    __typename: 'Category'
                  },
                  {
                    name: 'Backflow Preventer Kits',
                    __typename: 'Category'
                  },
                  {
                    name: 'Air Gaps',
                    __typename: 'Category'
                  },
                  {
                    name: 'Backflow Preventer Accessories',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Water Treatment',
                children: [
                  {
                    name: 'Water Filters',
                    __typename: 'Category'
                  },
                  {
                    name: 'Water Filter Parts & Accessories',
                    __typename: 'Category'
                  },
                  {
                    name: 'Water Softener Parts & Accessories',
                    __typename: 'Category'
                  },
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Water Softeners',
                    __typename: 'Category'
                  },
                  {
                    name: 'CATCLEAN',
                    __typename: 'Category'
                  },
                  {
                    name: 'Water Conditioners',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Water Heaters - Residential',
                children: [
                  {
                    name: 'Gas',
                    __typename: 'Category'
                  },
                  {
                    name: 'Electric',
                    __typename: 'Category'
                  },
                  {
                    name: 'Heat Pump',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              },
              {
                name: 'Water Heaters - Commercial',
                children: [
                  {
                    name: 'Electric',
                    __typename: 'Category'
                  },
                  {
                    name: 'Gas',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              }
            ],
            __typename: 'Category'
          },
          {
            name: 'TBC',
            children: [
              {
                name: 'TBC',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  },
                  {
                    name: 'Sheet Metal Pipe & Fittings',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              }
            ],
            __typename: 'Category'
          },
          {
            name: 'Faucets, Fixtures Appliances',
            children: [
              {
                name: 'TBC',
                children: [
                  {
                    name: 'TBC',
                    __typename: 'Category'
                  }
                ],
                __typename: 'Category'
              }
            ],
            __typename: 'Category'
          },
          {
            name: 'Faucets, FiXtures & Appliances',
            children: [],
            __typename: 'Category'
          }
        ],
        __typename: 'ProductCategories'
      },
      __typename: 'Query'
    }
  }
};

type MockCategory = {
  request: {
    query: DocumentNode;
    variables: {
      engines?: string;
    };
  };
  result: {
    data: ProductCategoriesQuery | undefined;
  };
};
