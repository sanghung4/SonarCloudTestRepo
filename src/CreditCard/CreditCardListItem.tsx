import {
  Box,
  Typography,
  TypographyProps
} from '@dialexa/reece-component-library';
import { useTranslation } from 'react-i18next';

import CreditCardTypeIcon from 'CreditCard/util/CreditCardTypeIcon';
import { CreditCard } from 'generated/graphql';
import { formatDate } from 'utils/dates';

type Props = {
  creditCard: CreditCard;
  expired: boolean;
  hideType?: boolean;
  hideDate?: boolean;
  hideExpired?: boolean;
  noEmphasis?: boolean;
  TypographyProps?: TypographyProps;
};

export default function CreditCardListItem(props: Props) {
  /**
   * Custom Hooks
   */
  const { t } = useTranslation();

  /**
   * Render
   */
  return (
    <Typography
      variant="body1"
      color="textSecondary"
      noWrap
      display="flex"
      alignItems="center"
      flexWrap="nowrap"
      {...props.TypographyProps}
    >
      {!props.hideType && (
        <CreditCardTypeIcon
          expired={props.expired}
          creditCard={props.creditCard}
        />
      )}
      <Box
        component="span"
        fontWeight={props.noEmphasis ? 400 : 700}
        ml={props.hideType ? 0 : 1}
        mr={props.hideDate ? 0 : 1}
      >
        {'**** '}
        {props.creditCard.creditCardNumber}
      </Box>{' '}
      {!props.hideDate &&
        formatDate(props.creditCard.expirationDate.date, 'MM/yy')}
      {!props.hideExpired && !!props.expired && (
        <Box ml={1} fontWeight={700} component="span" color="error.main">
          {t('creditCard.expired')}
        </Box>
      )}
    </Typography>
  );
}
