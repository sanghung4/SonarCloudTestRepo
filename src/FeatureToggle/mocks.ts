import { Feature } from 'generated/graphql';

export const mockFeatures: Feature[] = [
  { id: '1', isEnabled: false, name: 'noooo' },
  { id: '2', isEnabled: true, name: 'fine' },
  { id: '3', isEnabled: true, name: 'test' }
];
