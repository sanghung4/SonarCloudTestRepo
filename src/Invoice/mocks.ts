import { GetInvoiceDocument } from 'generated/graphql';

export const successMincron = {
  request: {
    query: GetInvoiceDocument,
    variables: {
      accountId: '123456',
      invoiceNumber: '5460026'
    }
  },
  result: {
    data: {
      invoice: {
        invoiceNumber: '5460026',
        status: 'Open',
        terms: 'NET 30 DAYS',
        customerPo: '40251/13325',
        invoiceDate: '10/28/2020',
        dueDate: '10/28/2020',
        originalAmt: '',
        openBalance: '',
        age: '',
        jobNumber: 'AVLNEP7',
        jobName: 'NEP 7 ASHEVILLE',
        address: {
          streetLineOne: '911 BLUEBERRY WAY',
          streetLineTwo: '',
          streetLineThree: '',
          city: 'NORTHLAKE',
          state: 'TX',
          postalCode: '76226',
          country: ''
        },
        shipDate: '11/06/2020',
        deliveryMethod: 'Our Truck',
        subtotal: '3184.28',
        tax: '262.76',
        otherCharges: '0.0',
        paidToDate: '1000.00',
        invoiceItems: [
          {
            id: '1738433F4-91A7-4637-8C60-B4DF59A060C1',
            brand: 'Advanced Drainage Systems, Inc.',
            name: '120 x 7.5, 100 Sq Yd, Geosynthetic, Double, Photodegradable, Erosion Control Blanket',
            partNumber: '8X1125STRAWDN',
            mfr: '00S2TT',
            thumb: 'null',
            price: '43.59',
            qty: {
              quantityOrdered: 1,
              quantityShipped: 1
            }
          }
        ]
      }
    }
  }
};

export const successMincron_5448540 = {
  request: {
    query: GetInvoiceDocument,
    variables: {
      accountId: '91492',
      invoiceNumber: '5448540'
    }
  },
  result: {
    data: {
      invoice: {
        invoiceNumber: '5282411',
        status: 'Open',
        terms: 'NET 30 DAYS',
        customerPo: 'WILKESBORO JIM',
        invoiceDate: '10/06/2021',
        dueDate: '10/28/2020',
        originalAmt: '',
        openBalance: '',
        age: '',
        jobNumber: 'SHOP',
        jobName: 'SHOP',
        address: {
          streetLineOne: '911 BLUEBERRY WAY',
          streetLineTwo: '',
          streetLineThree: '',
          city: 'NORTHLAKE',
          state: 'TX',
          postalCode: '76226',
          country: ''
        },
        shipDate: '11/06/2021',
        deliveryMethod: 'Our Truck',
        subtotal: '2184.28',
        tax: '162.73',
        otherCharges: '0.0',
        paidToDate: '139.91',
        invoiceItems: {
          id: 'F02FD85F-4DCE-4EA7-8459-2DA7CFD689DC',
          brand: 'Certex USA, name:2 x 12, Nylon, Eye and Eye, Ply Sling',
          partNumber: 'SLING212,mfr:CX07-0126-12',
          thumb:
            'https://images.tradeservice.com/ProductImages/DIR100193/CERTEX_CX07-0126-12_LINE_SML.jpg',
          price: '40.00',
          qty: {
            quantityOrdered: 1,
            quantityShipped: 1
          }
        }
      }
    }
  }
};
