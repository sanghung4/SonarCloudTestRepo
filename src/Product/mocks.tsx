import {
  GetProductDocument,
  GetProductPricingDocument
} from 'generated/graphql';

export const success = {
  request: {
    query: GetProductDocument,
    variables: {
      productInput: { productId: 'MSC-33860', customerNumber: undefined }
    }
  },
  result: {
    data: {
      product: {
        id: 'MSC-33860',
        partNumber: '33860',
        name: '1/2" Wrot Copper Cap',
        manufacturerName: 'Generic',
        manufacturerNumber: 'CCAPD',
        price: 0,
        stock: null,
        taxonomy: null,
        categories: [
          'Pipe & Fittings',
          'Metal Pipe & Fittings',
          'Copper Tube & Fittings'
        ],
        technicalDocuments: [
          {
            name: 'MFR Catalog Page',
            url: 'http://images.tradeservice.com/ATTACHMENTS/DIR100145/MUELERE00056_18_20.pdf',
            __typename: 'TechDoc'
          },
          {
            name: 'Technical Specification',
            url: 'http://images.tradeservice.com/ATTACHMENTS/DIR100240/MUELERE03088_1.pdf',
            __typename: 'TechDoc'
          },
          {
            name: 'MFR Item Data',
            url: 'http://images.tradeservice.com/ATTACHMENTS/DIR100145/MUELERE00601_1.pdf',
            __typename: 'TechDoc'
          }
        ],
        environmentalOptions: [],
        upc: '68576823241',
        unspsc: '40183104',
        seriesModelFigureNumber: null,
        productOverview:
          'Cap Fitting; Type Round Head; Nominal Size 1/2"; End Connection Female Soldered; Material Copper; Process Wrot; Application Ground Potable Water Supply System; Applicable Standard ASME B16.22, ANSI/NSF 61 G; Packaging Quantity 50 per Inner Pack, 1000 per Master Carton',
        featuresAndBenefits:
          'Copper Solder-Joint Fittings for Supply/Pressurized Systems have Been the Leading Brand of Copper Fittings for Over 80 Years; Available in Both Wrot Copper and Cast Bronze; The Acknowledged Experts at Engineering and Manufacturing Precision Solder-Joint Copper Fittings; Quality, Consistency and Reliability have Made the Streamline Brand Trusted and Specified All Around the World',
        techSpecifications: [
          {
            name: 'material',
            value: 'Copper',
            __typename: 'TechSpec'
          },
          {
            name: 'Applicable Standard',
            value: 'ASME B16.22, ANSI/NSF 61 G',
            __typename: 'TechSpec'
          },
          {
            name: 'Application',
            value: 'Ground Potable Water Supply System',
            __typename: 'TechSpec'
          },
          {
            name: 'End Connection',
            value: 'Female Soldered',
            __typename: 'TechSpec'
          },
          {
            name: 'Material',
            value: 'Copper',
            __typename: 'TechSpec'
          },
          {
            name: 'Nominal Size',
            value: '1/2"',
            __typename: 'TechSpec'
          },
          {
            name: 'Process',
            value: 'Wrot',
            __typename: 'TechSpec'
          },
          {
            name: 'Type',
            value: 'Round Head',
            __typename: 'TechSpec'
          }
        ],
        imageUrls: {
          thumb:
            'http://images.tradeservice.com/ProductImages/DIR100103/MUELER_W-07004_SML.jpg',
          small:
            'http://images.tradeservice.com/ProductImages/DIR100103/MUELER_W-07004_SML.jpg',
          medium:
            'http://images.tradeservice.com/ProductImages/DIR100103/MUELER_W-07004_MED.jpg',
          large:
            'http://images.tradeservice.com/ProductImages/DIR100103/MUELER_W-07004_LRG.jpg',
          __typename: 'ImageUrls'
        },
        packageDimensions: null,
        minIncrementQty: 0,
        erp: 'ECLIPSE',
        __typename: 'Product'
      }
    }
  }
};

export const pricing = {
  request: {
    query: GetProductPricingDocument,
    variables: {
      input: {
        customerId: '',
        branchId: 'any',
        productIds: ['33860'],
        includeListData: false
      }
    }
  },
  result: {
    data: {
      productPricing: {
        customerId: 'testAccount',
        branch: 'any',
        products: [
          {
            productId: '33860',
            orderUom: 'ea',
            catalogId: '',
            branchAvailableQty: 2,
            totalAvailableQty: 3,
            listIds: [],
            sellPrice: 3.256
          }
        ]
      }
    }
  }
};
