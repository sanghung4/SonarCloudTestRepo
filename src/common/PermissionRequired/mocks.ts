import { UserContextType } from 'AuthProvider';
import { Permission } from 'common/PermissionRequired';

export const authConfigNoPerm = {
  authState: {
    isAuthenticated: true
  },
  profile: {
    isEmployee: true,
    isVerified: true
  } as UserContextType['profile'],
  activeFeatures: ['LISTS']
};
export const authConfig = {
  authState: {
    isAuthenticated: true
  },
  profile: {
    userId: 'test',
    permissions: [Permission.EDIT_LIST, Permission.APPROVE_CART],
    isEmployee: true,
    isVerified: true
  } as UserContextType['profile'],
  activeFeatures: ['LISTS']
};
