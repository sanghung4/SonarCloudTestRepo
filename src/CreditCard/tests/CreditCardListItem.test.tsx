import CreditCardListItem from 'CreditCard/CreditCardListItem';
import { mockCreditCard } from 'CreditCard/tests/mocks/index.mocks';
import { render } from 'test-utils/TestWrapper';

describe('CreditCard - CreditCardListItem', () => {
  it('expect to match snapshot with minimum props', () => {
    const { container } = render(
      <CreditCardListItem creditCard={mockCreditCard} expired={false} />
    );
    expect(container).toMatchSnapshot();
  });

  it('expect to match snapshot as expired', () => {
    const { container } = render(
      <CreditCardListItem creditCard={mockCreditCard} expired />
    );
    expect(container).toMatchSnapshot();
  });

  it('expect to match snapshot to hide everything', () => {
    const { container } = render(
      <CreditCardListItem
        creditCard={mockCreditCard}
        expired={false}
        hideType
        hideDate
        hideExpired
        noEmphasis
      />
    );
    expect(container).toMatchSnapshot();
  });
});
