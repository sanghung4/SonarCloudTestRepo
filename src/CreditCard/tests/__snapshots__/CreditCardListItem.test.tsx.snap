// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreditCard - CreditCardListItem expect to match snapshot as expired 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-1ph717y-MuiTypography-root"
  >
    <svg
      class="MuiBox-root css-6rlkhh"
      data-testId="Visa"
      filter="grayscale(100%)"
      sx="[object Object]"
    >
      cc_visa.svg
    </svg>
    <span
      class="MuiBox-root css-8i1y65"
    >
      **** 
      1234567890123456
    </span>
     
    10/28
    <span
      class="MuiBox-root css-uwe8ff"
    >
      Expired
    </span>
  </p>
</div>
`;

exports[`CreditCard - CreditCardListItem expect to match snapshot to hide everything 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-1ph717y-MuiTypography-root"
  >
    <span
      class="MuiBox-root css-1a338f9"
    >
      **** 
      1234567890123456
    </span>
     
  </p>
</div>
`;

exports[`CreditCard - CreditCardListItem expect to match snapshot with minimum props 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-1ph717y-MuiTypography-root"
  >
    <svg
      class="MuiBox-root css-6rlkhh"
      data-testId="Visa"
      sx="[object Object]"
    >
      cc_visa.svg
    </svg>
    <span
      class="MuiBox-root css-8i1y65"
    >
      **** 
      1234567890123456
    </span>
     
    10/28
  </p>
</div>
`;
